server/jetstream_cluster.go

```diff
@@ -1918,6 +1936,207 @@ func (s *Server) jsClusteredStreamPurgeRequest(ci *ClientInfo, stream, subject,
 	n.Propose(encodeStreamPurge(sp))
 }
 
+// This will do a scatter and gather operation for all streams for this account.
+// This will be running in a separate Go routine.
+func (s *Server) jsClusteredStreamListRequest(acc *Account, ci *ClientInfo, offset int, subject, reply string, rmsg []byte) {
+	defer s.grWG.Done()
+
+	js, cc := s.getJetStreamCluster()
+	if js == nil || cc == nil {
+		return
+	}
+
+	js.mu.Lock()
+	defer js.mu.Unlock()
+
+	var streams []*streamAssignment
+	for _, sa := range cc.streams[acc.Name] {
+		streams = append(streams, sa)
+	}
+	// Needs to be sorted.
+	if len(streams) > 1 {
+		sort.Slice(streams, func(i, j int) bool {
+			return strings.Compare(streams[i].Config.Name, streams[j].Config.Name) < 0
+		})
+	}
+
+	scnt := len(streams)
+	if offset > scnt {
+		offset = scnt
+	}
+	if offset > 0 {
+		streams = streams[offset:]
+	}
+	if len(streams) > JSApiListLimit {
+		streams = streams[:JSApiListLimit]
+	}
+
+	rc := make(chan *StreamInfo, len(streams))
+
+	// Create an inbox for our responses and send out requests.
+	inbox := infoReplySubject()
+	rsub, _ := s.systemSubscribe(inbox, _EMPTY_, false, cc.c, func(_ *subscription, _ *client, _, reply string, msg []byte) {
+		var si StreamInfo
+		if err := json.Unmarshal(msg, &si); err != nil {
+			s.Warnf("Error unmarshaling clustered stream info response:%v", err)
+			return
+		}
+		select {
+		case rc <- &si:
+		default:
+			s.Warnf("Failed placing stream info result on internal chan")
+		}
+	})
+	defer s.sysUnsubscribe(rsub)
+
+	// Send out our requests here.
+	for _, sa := range streams {
+		isubj := fmt.Sprintf("$JSC.SI.%s.%s", sa.Client.Account, sa.Config.Name)
```

Comment:
Should "$JSC.SI.." made it a constant (like the template that we have)?

Reply:
Yes will do that.